src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/bulkscanprocessorclient/client/BulkScanProcessorClient.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.client;
+
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.exceptions.ReadEnvelopeException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.model.Envelope;
+
+import java.util.function.Supplier;
+
+@Service
+public class BulkScanProcessorClient {
+
+    private final RestTemplate restTemplate = new RestTemplate();
+    private final String baseUrl;
+    private final Supplier<String> serviceAuthTokenSupplier;
+
+    public BulkScanProcessorClient(
+        String baseUrl,
+        Supplier<String> serviceAuthTokenSupplier
+    ) {
+        this.baseUrl = baseUrl;
+        this.serviceAuthTokenSupplier = serviceAuthTokenSupplier;
+    }
+
+    public Envelope getEnvelopeById(String id) {
```

Comment:
Can we start introducing hystrix? any client call like this is a perfect example of the benefit of hystrix circuit breaker